python/lsst/ip/isr/isrFunctions.py

```diff
@@ -592,6 +594,230 @@ def brighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain, ga
+            ySelPixels = yc[sel]
+            # and add the flux into the pixel to the right or top
+            # depending on which axis we are handling
+            if ax == 0:
+                xPix = xSelPixels
+                yPix = ySelPixels+1
+            else:
+                xPix = xSelPixels+1
+                yPix = ySelPixels
+            # define flux as the either current pixel value or pixel
+            # above/right
+            # depending on whether positive or negative gradient
+            if i == 0:
+                # positive gradients, flux flowing to higher coordinate values
+                flux = factor * fStep[sel]*gx[sel]
+            else:
+                # negative gradients, flux flowing to lower coordinate values
+                flux = factor * fStep[yPix, xPix]*gx[sel]
+            # change the fluxes of the donor and receiving pixels
+            # such that flux is conserved
+            corr[sel] -= flux
+            corr[yPix, xPix] += flux
+
+    # return correction array
+    return corr
+
+
+def fluxConservingBrighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain,
+                                           gains=None, correctionMode=True):
+    """Apply brighter fatter correction in place for the image.
+    Modified version conserves flux, with improved correction of PSF core.
+    Also modified convolution to mitigate edge effects.
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to have brighter-fatter correction applied.  Modified
+        by this method.
+    kernel : `numpy.ndarray`
+        Brighter-fatter kernel to apply.
+    maxIter : scalar
+        Number of correction iterations to run.
+    threshold : scalar
+        Convergence threshold in terms of the sum of absolute
+        deviations between an iteration and the previous one.
+    applyGain : `Bool`
+        If True, then the exposure values are scaled by the gain prior
+        to correction.
+    gains : `dict` [`str`, `float`]
+        A dictionary, keyed by amplifier name, of the gains to use.
+        If gains is None, the nominal gains in the amplifier object are used.
+    correctionMode : `Bool`
+        If True (default) the function applies correction for BFE.  If False,
+        the code can instead be used to generate a simulation of BFE (sign
+        change in the direction of the effect)
+
+    Returns
+    -------
+    diff : `float`
+        Final difference between iterations achieved in correction.
+    iteration : `int`
+        Number of iterations used to calculate correction.
+
+    Notes
+    -----
+    Modified version of brighterFatterCorrection()
+    This correction takes a kernel that has been derived from flat
+    field images to redistribute the charge.  The gradient of the
+    kernel is the deflection field due to the accumulated charge.
+    Given the original image I(x) and the kernel K(x) we can compute
```

Comment:
The original blank lines are required for this to format correctly in the auto-generated documentation.  New lines before "Given", "Ic(x)", "Improved", "Because", and "Edges."